src/modules/txbuilder/RadixTransactionBuilder.ts

```diff
@@ -733,6 +786,51 @@ export default class RadixTransactionBuilder {
         return this
     }
 
+    // system param will be of type AcccountSystem when getState has been implemented everywhere.
+    /*
+    private setState(state: LedgerState, account: RadixAccount) {
+        let newState = {}
+        const address = account.getAddress()
+
+        newState[address] = state ? { ...state[address] } : createInitialState()
+
+        Object.keys(newState[address]).forEach((key) => {
+            account.accountSystems.forEach((system, name) => {
+                // Temporarily only use the systems that have been refactored so far
+                if(['TRANSFER','TOKENS'].includes(system.name)) {
+                    newState[address][key] = newState[address][key] ? newState[address][key] : system.getState()[key]
+                }
+            })
+        })
+        return newState
+    }
+    */
+
+    private addAction(account: RadixAccount, action: Function) {
```

Comment:
We should stay consistent with the java library and call it staging and action. Also this could definitely be public if the action type was better defined.

Reply:
Renamed it.
So you mean, that we should support custom actions?
That makes sense. Should we support this now?

Reply:
To properly support it I think there will need to be more work done in structure and in documentation, but no reason to not make the method public in case anyone knows what they're doing

Reply:
Agreed. Fixed.